src/tests/fs_tests.rs

```diff
@@ -3,6 +3,18 @@ mod fs_tests {
     use crate::interface;
     use crate::safeposix::{cage::*, filesystem};
     use super::super::*;
+
+    #[test]
+    pub fn persistancetest() {
+        let cage = init_cage();
+
+        let fd = cage.open_syscall("/testfile", O_CREAT | O_EXCL | O_RDWR, S_IRWXA);
+        assert!(fd >= 0);
+
+        filesystem::persist_metadata();
+        filesystem::restore_metadata();
+    }
```

Comment:
Maybe do the close and unlink and then the open/access to check if it's still there in your test to really test it's deserializing right?

Reply:
I do think we need more tests here but we I think we need more syscalls implemented to do that fully.